storage/src/nodestore/mod.rs

```diff
@@ -330,12 +330,23 @@ where
     fn root_node(&self) -> Option<SharedNode> {
         self.deref().root_node()
     }
+    fn root_as_maybe_persisted_node(&self) -> Option<MaybePersistedNode> {
+        self.deref().root_as_maybe_persisted_node()
+    }
 }
 
 /// Reads the root of a merkle trie.
+///
+/// The root may be None if the trie is empty.
 pub trait RootReader {
     /// Returns the root of the trie.
+    /// Callers that just need the node at the root should use this function.
     fn root_node(&self) -> Option<SharedNode>;
+
+    /// Returns the root of the trie as a `MaybePersistedNode`.
+    /// Callers that might want to modify the root or know how it is stored
+    /// should use this function.
+    fn root_as_maybe_persisted_node(&self) -> Option<MaybePersistedNode>;
```

Comment:
In a future diff, this will just be called `root` but I want it verbose during the rest of the development of this feature.